nilearn/image/image.py

```diff
@@ -638,6 +641,9 @@ def threshold_img(img, threshold, mask_img=None):
     img_data = _safe_get_data(img).copy()
     affine = img.get_affine()
 
+    if np.isnan(np.sum(img_data)):
```

Comment:
That's a great comment. Actually, when doing this we could remove the "copy" above, as nan_to_num always does a copy.